frontend/appflowy_flutter/lib/mobile/presentation/database/card/row/cells/date_cell/widgets/date_and_time_display.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:appflowy/plugins/database_view/widgets/row/cells/date_cell/date_cal_bloc.dart';
+import 'package:appflowy_backend/protobuf/flowy-database2/date_entities.pbenum.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flowy_infra_ui/widget/spacing.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class DateAndTimeDisplay extends StatelessWidget {
+  const DateAndTimeDisplay(this.state, {super.key});
+  final DateCellCalendarState state;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        // date/start date and time
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            _DateEditButton(
+              dateStr: state.isRange
+                  ? _formatDateByDateFormatPB(
+                      state.startDay,
+                      state.dateTypeOptionPB.dateFormat,
+                    )
+                  : state.dateStr,
+              initialDate: state.isRange ? state.startDay : state.dateTime,
+              onDaySelected: (newDate) {
+                if (state.isRange) {
+                  context
+                      .read<DateCellCalendarBloc>()
+                      .add(DateCellCalendarEvent.setStartDay(newDate));
+                } else {
+                  context.read<DateCellCalendarBloc>().add(
+                        DateCellCalendarEvent.selectDay(newDate),
+                      );
+                }
```

Comment:
Same here, personal pref, formatting is off for the suggestion tho
```suggestion
                context.read<DateCellCalendarBloc>()
                  .add(
                    state.isRange
                        ? DateCellCalendarEvent.setStartDay(newDate)
                        : DateCellCalendarEvent.selectDay(newDate),
                  );
```